libnmstate/netapplier.py

```diff
@@ -313,12 +315,20 @@ def assert_ifaces_state(ifaces_desired_state, ifaces_current_state):
             ifaces_desired_state[ifname], iface_cstate)
 
         _cleanup_iface_ethernet_state_sanitize(iface_dstate, iface_cstate)
+        _sort_ip_addresses(iface_dstate)
 
         if iface_dstate != iface_cstate:
             raise DesiredStateIsNotCurrentError(
                 'desired: {}\ncurrent: {}'.format(iface_dstate, iface_cstate))
 
 
+def _sort_ip_addresses(iface):
```

Comment:
I remember that @thom311 told me that the order of IP addresses matters but I forgot for what exactly. Maybe for the default IP address that is used for outgoing connections via that interface.